src/rdrand.rs

```diff
@@ -33,12 +37,15 @@ unsafe fn rdrand() -> Result<[u8; WORD_SIZE], Error> {
             if el != 0 && el != !0 {
                 return Ok(el.to_ne_bytes());
             }
-            error!("RDRAND returned {:X}, CPU RNG may be broken", el);
+            is_bad = true;
             // Keep looping in case this was a false positive.
         }
     }
-    error!("RDRAND failed, CPU issue likely");
-    Err(Error::UNKNOWN)
+    Err(Error::from_custom_error(if is_bad {
```

Comment:
Why (1) is reporting distinct errors if `is_bad` useful, and (2) do we not want to log via `error!`?

Reply:
This was a response to: https://github.com/rust-random/getrandom/pull/48#pullrequestreview-256038770, but I now think returning different error codes is more confusing rather than less confusing. Now we just `error!` log suspicious values and return `FAILED_RDRAND` in both cases.